File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left=0,right=height.length-1;
        int max_area=0;
        int current=0;
        while(left<right){
            current=Math.min(height[left],height[right])*(right-left);
            max_area=Math.max(current,max_area);
            if(height[left]<height[right]){
                left++;
            }
            else{
                right--;
            }
        }
        return max_area;
    }
}

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l,r=0,len(height)-1
        max_area=0
        while(l<r):
            width=r-l
            h=min(height[l],height[r])
            area=width*h
            max_area=max(area,max_area)
            if height[l]<height[r]:
                l+=1
            elif height[l]==height[r]:
                l+=1
            else:
                r-=1
        return max_area